### PR TITLE
chore(build): record preferred NPM publishing order

### LINKED ISSUE #13221: (pipelines): CDK packages deployment race condition
When using `pipelines.CdkPipeline`, the default is to install the latest CDK version, which for me was 1.91.0, which happened to have been deployed only minutes before. But installation failed because it then tried to install `@aws-cdk/cloud-assembly-schema` of the same version, which was not yet available on npm.

Error I got in the selfupdate stage in CodeBuild:
```
[Container] 2021/02/23 13:23:35 Entering phase INSTALL
[Container] 2021/02/23 13:23:35 Running command npm install -g aws-cdk
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @aws-cdk/cloud-assembly-schema@1.91.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-02-23T13_23_40_901Z-debug.log
[Container] 2021/02/23 13:23:40 Command did not exit successfully npm install -g aws-cdk exit status 1
[Container] 2021/02/23 13:23:40 Phase complete: INSTALL State: FAILED
[Container] 2021/02/23 13:23:40 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: npm install -g aws-cdk. Reason: exit status 1
```

I think that the actual fix for this is for me to set `cdkCliVersion` to a version I know is on npm, but this was still surprising behaviour.

Things I can think of that might fix this:
- If cdk version X depends on  cloud-assembly-schema version X, should cloud-assembly-schema get released first?
- Alternatively, it might make sense for `CdkPipeline` to default to the version of CDK that is in use (so the version used _within_ the pipeline matches the version used to _deploy_ the pipeline).

### PR DESCRIPTION
**Pull request by rix0rrr**
Use lerna to generate a topologically sorted list of `.tgz` files.

This list will be used during publishing to impose an order
on the package publishing actions.

Fixes #13221.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: pack.sh [hunk 1]
```diff
@@ -63,6 +63,15 @@ for dir in $(find packages -name dist | grep -v node_modules | grep -v run-wrapp
   rsync -a $dir/ ${distdir}/
 done
 
+# Record the dependency order of NPM packages into a file
+# (This file will be opportunistically used during publishing, it does not need to be complete)
```
Comment (nija-at) [inline pack.sh:67]: Do the ones not in this file go before or after or random? Nice to include that in the comment.

File: pack.sh [hunk 2]
```diff
@@ -63,6 +63,15 @@ for dir in $(find packages -name dist | grep -v node_modules | grep -v run-wrapp
...
+node_modules/.bin/lerna exec \
```
Comment (nija-at) [inline pack.sh:68]: why not `lerna ls --toposort`? Isn't that what that's meant for?

File: pack.sh (hunk 1 above)
Comment (rix0rrr) [reply pack.sh:67]: Unspecified, do not rely on any behavior there.

The details of how everything works will be documented in the PR that introduces the feature in aws-delivlib.

File: pack.sh [hunk 3]
```diff
@@ -36,6 +36,8 @@ function lerna_scopes() {
   done
 }
 
+if false; then
+
```
Comment (nija-at) [inline pack.sh:40]: err.... from your local testing?

**Review by nija-at — APPROVED** [review]

**Pull request merged**